### ISSUE TITLE
Infinite loop

### ISSUE THREAD

**Issue by atik-lab** [recws-org/recws#31]
In got an infinite loop once a day, after profiling the infinite loop is here at around line 375.

```go
		for {
			if !rc.IsConnected() {
				continue
			}
```

It doesn't go away until I restart my application.

**Comment by loeffel-io** [comment]
thanks @atik-lab, i think this belongs to #29

**Comment by loeffel-io** [comment]
could you please try the v1.2.1

**Comment by atik-lab** [comment]
Sure, I will let you know if it happened again. In my opinion a small sleep would be good to add there, just to not burn the CPU.

**Comment by atik-lab** [comment]
It still get's disconnected and does not connect again and it makes an infinite loop. I will try now with the 1.2.2 and the following code:

```
			if !rc.IsConnected() {
				rc.CloseAndReconnect()
				return
			}
```

**Comment by atik-lab** [comment]
Hello, I found the error, I was using a module that uses your module and when `messageType, data, err := b.conn.ReadMessage()` had an error it returned without closing properly. So it is not really your module fault but if not well used it can be really annoying to have that infinite for loop.
Please feel free to close the issue.

**Comment by loeffel-io** [comment]
For now this is solved by `v1.2.3`
